lucene/queries/src/java/org/apache/lucene/queries/function/FunctionValues.java

```diff
@@ -151,6 +164,11 @@ public ValueSourceScorer getScorer(Weight weight, LeafReaderContext readerContex
       public boolean matches(int doc) {
         return true;
       }
+      @Override
+      public float costEvaluationFunction() {
+        // Match everything
```

Comment:
Nitpick comment: the point isn't that we're matching everything, the point is that matches() merely returns a constant and is thus virtually free.

Reply:
Removed the comment, thanks